### ISSUE TITLE
Rename states with "Mode" in the name "State" instead

### ISSUE THREAD

**Issue by andreas-junghanns** [modelica/fmi-standard#718]
We have a number of states in our state diagram, that are called "Something Mode".

To make matters worse, we also call the different kinds of Co-Simulation "Modes": 
. Basic Co-Simulation
. Hybrid Co-Simulation
. Scheduled Execution Simulation

I suggest we use "state" instead of "mode" when we name states.

**Comment by andreas-junghanns** [comment]
And we also talk about "debug mode", "batch mode", "interactive mode", "evaluation mode", "subactive mode", "asynchronous mode"...

**Comment by chrbertsch** [comment]
Regular Design Meeting:
Torsten S: "Asynchronous State" does not sound good. I would stick with mode
Andreas: J: I agree that mode sounds better.
Andreas J: The long version would be "Asynchronous Simulation State", is too long ...
Andreas P, Torsten: We would have to change the API names also  ("enterEventMode")
Torsten S: I see no benefit
Andreas J: It would bring clarity 
Christian, Andreas J: We could also rename the "modes" of HybridCosimulation. "Variants"? 
Torsten: They belong to "Interface Types"
Christian: Could we remove "mode" when talking about e.g. "Scheduled Execution" ...
Andreas J: I would use "Cosimulation Types" (Interface Type has a meaning in Java)


Andreas: I found much more ... "Debug mode"


Closing this ticket: We do not want to change the API, create a new ticket to renamce the co-simulation types, see #724
